build_tools/provision_machine.py

```diff
@@ -275,6 +297,12 @@ def main(argv):
         help="Github release version of TheRock to provision, from the nightly-release (X.Y.ZrcYYYYMMDD) or dev-release (X.Y.Z.dev0+{hash})",
     )
```

Comment:
Can't comment on the specific line and unrelated to this PR, but be careful with permalinks in help text. 

```python
    parser.add_argument(
        "--amdgpu-family",
        type=str,
        default="gfx94X-dcgpu",
        help="AMD GPU family to provision (please refer to this: https://github.com/ROCm/TheRock/blob/59c324a759e8ccdfe5a56e0ebe72a13ffbc04c1f/cmake/therock_amdgpu_targets.cmake#L44-L81 for family choices)",
    )
```

That ref can get out of date, while a more stable URL like https://github.com/ROCm/TheRock/blob/main/cmake/therock_amdgpu_targets.cmake, https://llvm.org/docs/AMDGPUUsage.html#processors , or a markdown or wiki file somewhere in this repository could be kept up to date.